keps/sig-scheduling/1923-try-nominated-node-first/README.md

```diff
@@ -0,0 +1,224 @@
+- [ ] (R) Production readiness review completed
+- [ ] Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+If the incoming pod that going to be scheduled has the priority defined,
+and there are no nodes in the cluster that satisfy the scheduling filters
+for the incoming Pod, scheduler will try to preempt lower priority pods
+running on a selected node in the cluster to make room for the
+pod to be scheduled, the selected node for the incoming pod after the
+preemption will be set as the value of `pod.Status.NominatedNodeName`.
+
+The Node is called *Nominated* to indicate the intent for the Pod to be
+scheduled on it once preemption of other Pods finish. However, the
+`Pod.status.nominatedNodeName` information is not directly used in the
+following scheduling cycles.
+
+Pod scheduling is split into two phases, the scheduling cycle and the binding
+cycle, the scheduling cycle primarily includes filtering and scoring.
+
+When the preemption happened in a previous scheduling cycle, there is a high
+chance that the nominated node is the *only* node that satisfies the filters
+for the incoming Pod to be scheduled.
+
+This KEP proposes to validate the availability of the nominated node in the
+new scheduling cycle before looping all over the nodes in the cluster for the
+phase of the filtering.
+
+## Motivation
+
+In real production environment, pods can have different priorites due to business
+needs, the preemption could happen to make sure high priority pods could got scheduled
+and spawned.
+
+For the cluster with huge number of computing nodes, looping over all the nodes in
+the cluster for the phase of the filtering is an unnegligible consuming for scheduler.
+
+### Goals
+
+- In the case of `pod.Status.NominatedNodeName` is set for the incoming pod, scheduler
+  will try the nominated node first for the filtering phase, failed to schedule on the
+  nominated node will fall back to verify all the nodes in the cluster instead of preemption
+  again.
+
+## Proposal
+
+### User Stories (Optional)
+
+As a premium user, I want my high priority application got spin up more quickly,
+I don't care where the application is running, whether there is any other nodes
+in the cluster have more disk space, memory size etc., as long as my application
+run normally.
```

Comment:
I would phrase this as follows: Users want faster scheduling. Since it is highly likely the pod will only fit on the nominated node, the improvement in scheduling latency will come at negligible cost (the cost being placing the pod on a less optimal node).

Reply:
Done